cmd/site-replication.go

```diff
@@ -647,6 +650,76 @@ func (c *SiteReplicationSys) validateIDPSettings(ctx context.Context, peers []Pe
 	return true, nil
 }
 
+// Netperf for site-replication net perf
+func (c *SiteReplicationSys) Netperf(ctx context.Context, duration time.Duration) (results madmin.SiteNetPerfResult, err error) {
+	infos, err := globalSiteReplicationSys.GetClusterInfo(ctx)
+	if err != nil {
+		return results, err
+	}
+	var wg sync.WaitGroup
+	var resultsMu sync.RWMutex
+	for _, info := range infos.Sites {
+		info := info
+		// will call siteNetperf, means call others's adminAPISiteReplicationDevNull
+		if globalDeploymentID == info.DeploymentID {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				cli, err := globalSiteReplicationSys.getAdminClient(ctx, info.DeploymentID)
+				if err != nil {
+					return
+				}
+				result := siteNetperf(ctx, duration)
```

Comment:
pass the endpoint so that you can fill it up in the result instead of setting it here.